packages/visualizations/src/components/Legend/CategoryLegend.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+    import type { CategoryLegend } from './types';
+
+    export let legendOptions: CategoryLegend;
+    let refinedSeries: string[] = [];
+    // if (legendOptions.type === 'category') {
+    //     refinedSeries = legendOptions.items.map(() => false);
+    // }
+</script>
+
+{#each legendOptions.items as item, i}
+    <div
+        class="color-item-category"
+        on:click={() => {
+            // using color as a unique id proxy
+            if (item?.color) {
+                const isItemRefined = Boolean(refinedSeries.find((color) => color === item.color));
+                refinedSeries = isItemRefined
+                    ? refinedSeries.filter((color) => color !== item.color)
+                    : [...refinedSeries, item.color];
+            }
+            item.onClick(i);
+        }}
+        on:mouseenter={() => {
+            if (item.onHover) {
+                item.onHover(i);
+            }
+        }}
+        on:mouseleave={() => {
+            if (item.onLeave) {
+                item.onLeave();
+            }
+        }}
+    >
+        {#if item.color}
+            <div
+                class="color-box-category"
+                class:refined={refinedSeries[i]}
```

Comment:
```suggestion
                class:refined={Boolean(refinedSeries.find((color) => color === item.color))}
```
It doesn't seem safe to use index here, while testing it appears that it can easily not match the refined array (tested in storybook)

Reply:
![Capture d’écran 2023-05-31 à 17 38 30](https://github.com/opendatasoft/ods-dataviz-sdk/assets/84501475/a0f3675e-3c2a-4058-8cf7-2435e9e1e503)

To reproduce the unwanted behaviour, if you click first on the blue box then it disappears from the chart but the red box is marked as refined

Reply:
Ha yes, left over from the previous method 👌